### DISCUSSION TITLE
Cannot install curl on CentOS7

### DISCUSSION THREAD

**Discussion by neheb** [Homebrew/discussions#1176] [Linux] [answered]
The issue is that fontconfig uses gperf during its compilation. Unfortunately, the gperf binary is not linked correctly:

```
[mangix@a ~]$ ldd .linuxbrew/bin/gperf
/usr/bin/bash: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8)
        linux-vdso.so.1 (0x00007fff68580000)
        libstdc++.so.6 => not found
        libm.so.6 => /home/mangix/.linuxbrew/lib/libm.so.6 (0x00007fbbf1e74000)
        libgcc_s.so.1 => not found
        libc.so.6 => /home/mangix/.linuxbrew/lib/libc.so.6 (0x00007fbbf1cd6000)
        /home/mangix/.linuxbrew/lib/ld.so (0x00007fbbf1f76000)
```

Note the not found.

fontconfig is needed for curl.

**Comment by neheb** [comment, marked as answer]
I found the issue. Both of those symlinks are bad. They point to OS directories that don't exist.

```
lrwxrwxrwx. 1 mangix mangix   35 Oct 10  2017 libgcc_s.so.1 -> /lib/x86_64-linux-gnu/libgcc_s.so.1
lrwxrwxrwx. 1 mangix mangix   72 Oct 10  2017 libstdc++.so.6 -> /usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libstdc++.so.6
```

I have no idea how those paths were derived. CentOS 7 has GCC4, not 5.

Manually fixing those symlinks fixed the issue.

**Reply by neheb** [reply]
I will also note fontconfig has a meson build available. In case of no host gperf, it will build its own gperf.

**Reply by danielnachun** [reply]
You found the correct solution.  See here for more details: https://github.com/Homebrew/linuxbrew-core/issues/22511.

**Reply by neheb** [reply]
Yeah saw that after I posted this.
